examples_tests/tests.py

```diff
@@ -0,0 +1,180 @@
+            'dir': 'downloader-with-wiki-parts',
+            'snap': 'downloader_1.0_amd64.snap'
+         }),
+        ('godd', {
+            'dir': 'godd',
+            'snap': 'godd_1.0_amd64.snap'
+         }),
+        ('gopaste', {
+            'dir': 'gopaste',
+            'snap': 'gopaste_1.0_amd64.snap'
+         }),
+        ('java-hello-world', {
+            'dir': 'java-hello-world',
+            'snap': 'java-hello-world_0_amd64.snap'
+         }),
+        ('libpipeline', {
+            'dir': 'libpipeline',
+            'snap': 'pipelinetest_1.0_amd64.snap',
+            'internal_tests_commands': [
+                ('/apps/bin/pipelinetest.pipelinetest',
+                 'running ls | grep c\n'
+                 'custom libpipeline called\n'
+                 'include\n')],
+         }),
+        ('py2-project', {
+            'dir': 'py2-project',
+            'snap': 'spongeshaker_0_amd64.snap'
+         }),
+        ('py3-project', {
+            'dir': 'py3-project',
+            'snap': 'spongeshake_0_amd64.snap'
+         }),
+        ('qmldemo', {
+            'dir': 'qmldemo',
+            'snap': 'qmldemo_1_amd64.snap'
+         }),
+        ('ros', {
+            'dir': 'ros',
+            'snap': 'ros-example_1.0_amd64.snap'
+         }),
+        ('shout', {
+            'dir': 'shout',
+            'snap': 'shout_0.52.0_amd64.snap'
+         }),
+        ('tomcat-maven-webapp', {
+            'dir': 'tomcat-maven-webapp',
+            'snap': 'tomacat-webapp-demo_1.0_amd64.snap'
+         }),
+        ('webcam-webui', {
+            'dir': 'webcam-webui',
+            'snap': 'webcam-webui_1_amd64.snap'
+         }),
+        ('webchat', {
+            'dir': 'webchat',
+            'snap': 'webchat_0.0.1_amd64.snap'
+         }),
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        cls.temp_dir = tempfile.mkdtemp()
+        cls.image_path = os.path.join(cls.temp_dir, 'snappy.img')
+        logger.info('Creating a snappy image to run the tests.')
+        subprocess.check_call(
+            ['sudo', 'ubuntu-device-flash', '--verbose',
+             'core', 'rolling', '--channel', 'edge',
+             '--output', cls.image_path, '--developer-mode'])
+        logger.info('Running the snappy image in a virtual machine.')
+        system = subprocess.check_output(['uname', '-m']).strip().decode('utf8')
+        qemu_command = ('qemu-system-' + system +
+            ' -m 512 -nographic -net user -net nic,model=virtio' +
+            ' -drive file=' + cls.image_path +
+            ',if=virtio -redir tcp:{}::22'.format(cls.testbed_port) +
+            ' -monitor none -serial none')
+        cls.vm_process = subprocess.Popen(qemu_command, shell=True)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.vm_process.kill()
+        shutil.rmtree(cls.temp_dir)
+
+    def setUp(self):
+        super().setUp()
+        self._wait_for_ssh()
+
+    def _wait_for_ssh(self):
```

Comment:
python3-paramiko is not in trusty. We could install it with pip. Do you think it's worth it?